spec/mongo/server/description_spec.rb

```diff
@@ -431,19 +431,36 @@
         described_class.new(address, { 'ismaster' => false })
       end
 
-      it 'returns true' do
+      it 'returns false' do
         expect(description).to_not be_primary
       end
     end
 
     context 'when the server is a primary' do
 
-      let(:description) do
-        described_class.new(address, replica)
+      context 'when the hostname contains no capital letters' do
+
+        let(:description) do
+          described_class.new(address, replica)
+        end
+
+        it 'returns true' do
+          expect(description).to be_primary
+        end
       end
 
-      it 'returns false' do
-        expect(description).to be_primary
+      context 'when the hostname contains capital letters' do
+
+        let(:description) do
+          described_class.new('localhost:27017',
```

Comment:
Is there a test for this hostname having capital letters and ismaster response being in lowercase?

Reply:
The driver always converts addresses to lowercase before attempting to make a connection, so a ServerDescription would never have a non-lowercase address. The issue came up specifically because the ServerDescription was not changing the address in the primary host field of the isMaster response in the same way.

Reply:
OK then.